src/category_theory/equivalence.lean

```diff
@@ -203,6 +203,58 @@ by { dsimp [inv_fun_id_assoc], tidy }
   (inv_fun_id_assoc e F).inv.app X = F.map (e.counit_inv.app X) :=
 by { dsimp [inv_fun_id_assoc], tidy }
 
+
+
+section cancellation_lemmas
+variables (e : C ≌ D)
+
+-- We need special forms of `cancel_nat_iso_hom_right(_assoc)` and `cancel_nat_iso_inv_right(_assoc)`
+-- for units and counits, because the simplifier can't see that `(𝟭 C).obj X` is the same as `X`.
```

Comment:
Is `dsimp`  also not able to see this? Do you know why not?

Reply:
It seems I had misdiagnosed the need for these lemmas. It doesn't seem to be anything to do with `(𝟭 C).obj X`.

Instead, neither `simp` nor `rw` will rewrite along `cancel_nat_iso_hom_right` unless you provide an explicit argument, e.g. `simp [cancel_nat_iso_hom_right e.unit_iso]`. I'm not sure why. :-(

These additional lemmas do fire, however. I've updated the comment to reflect this.